### PR TITLE
[stable/jenkins] Add image pull secrets to jenkins templates

### PR DESCRIPTION
**Pull request by berndtj**
In order to use some private repositories, kubernetes needs docker
registry secrets to authenticate.  This adds the ability to set
the secrets for both the master and agent images.

### CODE REVIEW

File: stable/jenkins/templates/config.yaml [hunk 1]
```diff
@@ -69,7 +69,15 @@ data:
               </containers>
               <envVars/>
               <annotations/>
+{{- if .Values.Agent.ImagePullSecret }}
+              <imagePullSecrets>
+                <org.csanchez.jenkins.plugins.kubernetes.PodImagePullSecret>
+                  <name>{{ .Values.Agent.ImagePullSecret }}</name>
+                </org.csanchez.jenkins.plugins.kubernetes.PodImagePullSecret>
+              </imagePullSecrets>
```
Comment (lachie83) [inline stable/jenkins/templates/config.yaml:88]: Is this close correct? It looks different to L79?

File: stable/jenkins/templates/config.yaml [hunk 2]
```diff
@@ -69,7 +69,15 @@ data:
               </containers>
               <envVars/>
               <annotations/>
+{{- if .Values.Agent.ImagePullSecret }}
+              <imagePullSecrets>
```
Comment (lachie83) [inline stable/jenkins/templates/config.yaml:84]: Should this be outside the if block given else will just print a close?

File: stable/jenkins/templates/jenkins-master-deployment.yaml [hunk 3]
```diff
@@ -134,3 +134,7 @@ spec:
       {{- else }}
         emptyDir: {}
       {{- end -}}
+{{- if .Values.Master.ImagePullSecret }}
+      imagePullSecrets:
+      - name: {{ .Values.Master.ImagePullSecret }}
+{{- end -}}
```
Comment (lachie83) [inline stable/jenkins/templates/jenkins-master-deployment.yaml]: Nit: newline

**Review by lachie83 — DISMISSED** [review]
Please also bump the chart version

File: stable/jenkins/templates/config.yaml (hunk 2 above)
Comment (berndtj) [reply stable/jenkins/templates/config.yaml:84]: Yeah, that will remove some boilerplate.  I can.

File: stable/jenkins/templates/config.yaml (hunk 1 above)
Comment (berndtj) [reply stable/jenkins/templates/config.yaml:88]: It's correct, as the L79 close is a single line close.  Basically, I can remove it if I do as you indicate in the first comment.

File: stable/jenkins/templates/jenkins-master-deployment.yaml (hunk 3 above)
Comment (berndtj) [reply stable/jenkins/templates/jenkins-master-deployment.yaml]: sure :)

**Comment by viglesiasce** [comment]
Also implemented in https://github.com/kubernetes/charts/pull/1878

**Review by viglesiasce — APPROVED** [review]

**Pull request merged**